core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -259,19 +258,16 @@ && allNonNull(dataNullCounts, field)) {
   }
 
   private static <T> boolean rangesOverlap(
-      Type.PrimitiveType type,
+      Types.NestedField field,
       ByteBuffer dataLowerBuf,
       ByteBuffer dataUpperBuf,
-      ByteBuffer deleteLowerBuf,
-      ByteBuffer deleteUpperBuf) {
-    Comparator<T> comparator = Comparators.forType(type);
+      T deleteLower,
+      T deleteUpper) {
+    Type.PrimitiveType type = field.type().asPrimitiveType();
+    Comparator<T> cmp = Comparators.forType(type);
     T dataLower = Conversions.fromByteBuffer(type, dataLowerBuf);
```

Comment:
So, the most performance improvement comes here.

Reply:
Yep.